cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -484,6 +488,124 @@ public PackageSearchResult searchPackage(String query, String supportedPlatform,
+                                    error.getMessage());
+                        }
+                    }
+                }
+            }
+
+            throw new CentralClientException(ERR_CANNOT_GET_CONNECTOR + "'" + query + "'.");
+        } catch (IOException e) {
+            throw new CentralClientException(ERR_CANNOT_GET_CONNECTOR + "'" + query + "'. reason: " + e.getMessage());
+        } finally {
+            body.ifPresent(ResponseBody::close);
+            try {
+                this.closeClient(client);
+            } catch (IOException e) {
+                // ignore
+            }
+        }
+    }
+
+    /**
+     * Get connector by id.
+     */
+    public BalConnector getConnector(String id, String supportedPlatform, String ballerinaVersion)
+            throws CentralClientException {
+        Optional<ResponseBody> body = Optional.empty();
+        OkHttpClient client = this.getClient();
+        try {
+            Request searchReq = getNewRequest(supportedPlatform, ballerinaVersion)
+                    .get()
+                    .url(this.baseUrl + "/" + CONNECTORS + "/" + id)
+                    .build();
+
+            Call httpRequestCall = client.newCall(searchReq);
+            Response searchResponse = httpRequestCall.execute();
+
+            body = Optional.ofNullable(searchResponse.body());
+            if (body.isPresent()) {
+                Optional<MediaType> contentType = Optional.ofNullable(body.get().contentType());
+                if (contentType.isPresent()  && isApplicationJsonContentType(contentType.get().toString())) {
+                    // If searching was successful
+                    if (searchResponse.code() == HttpsURLConnection.HTTP_OK) {
+                        BalConnector connectorResult = new Gson().fromJson(body.get().string(),
+                                BalConnector.class);
+                        return connectorResult;
+                    }
+
+                    // If search request was sent wrongly
+                    if (searchResponse.code() == HttpsURLConnection.HTTP_BAD_REQUEST) {
+                        Error error = new Gson().fromJson(body.get().string(), Error.class);
+                        if (error.getMessage() != null && !"".equals(error.getMessage())) {
+                            throw new CentralClientException(error.getMessage());
+                        }
+                    }
+
+                    // If error occurred at remote repository
+                    if (searchResponse.code() == HttpsURLConnection.HTTP_INTERNAL_ERROR) {
+                        Error error = new Gson().fromJson(body.get().string(), Error.class);
+                        if (error.getMessage() != null && !"".equals(error.getMessage())) {
+                            throw new CentralClientException(ERR_CANNOT_GET_CONNECTOR + "'" + id + "' reason:" +
+                                    error.getMessage());
+                        }
+                    }
```

Comment:
Can we move this error handling to a separate method? It seems this same logic has been duplicated above as well.